### PR TITLE
Feature/line chart

### PR DESCRIPTION
**Pull request by capetillo**
## FEATURE: Line chart with fake data 

**Background:**
The next step to get the Line Profile is to get a chart to display the data associated with the line and deep sky objects. For now, I am displaying fake data. 

**Implementation:**
Using d3.js, I implemented a plot chart. The Y-axis is completely random and the X-axis is the length in pixels of the line drawn. These two change dynamically. 
In the store (rip), I added two states: `randomNumbers` and `lineLength` as well as two mutations for these.
In `ImageViewer`, I generate 10 random numbers between 0 and 65,000 and store these in the `randomNumbers` array. These numbers are generated when a line is drawn. 
In this same file, I calculate the length of the line in pixels by using the pythagorean theorem and store this value in the store. 
Lastly, in `LineChart` I implement the logic to initialize d3 and have the axes be updated as a new line is drawn. 

I also added a very small fix to `ImageViewer` that prevents the line from being drawn outside of the image. 

### VISUALS
**Check out the console. You can see that the length stays about the same as I draw the line between those two points. We also know that this image is 1000px x 1000px and that's the last value that we see. You can also see the Y-axis' values updating as a line gets drawn and at the end, both axes update**

https://github.com/LCOGT/datalab-ui/assets/54489472/0f6683ae-e28e-4159-8369-30a8467fd724

### CODE REVIEW

File: src/components/Project/ImageViewer.vue [hunk 1]
```diff
@@ -25,36 +28,46 @@ let imageBounds
...
 const loadImageOverlay = (src) => {
   const img = new Image()
   img.onload = () => {
     // Getting image bounds based on img's size
     imageBounds = [[0, 0], [img.height, img.width]] 
-
     if (imageOverlay) {
       image.removeLayer(imageOverlay)
     }
+    imageWidth.value = img.width
```
Comment (capetillo) [inline src/components/Project/ImageViewer.vue:43]: and this too

File: src/components/Project/ImageViewer.vue [hunk 2]
```diff
@@ -25,36 +28,46 @@ let imageBounds
...
+    imageHeight.value = img.height
     // Add new overlay with correct bounds
     imageOverlay = L.imageOverlay(src, imageBounds).addTo(image)
     // Fit image view to the bounds of the image
     image.fitBounds(imageBounds)
     image.setMaxBounds(imageBounds)
 
+    // waits for DOM to load then recalculates image's size and position after the container's dimensions have changed (built in method of Leaflet)
```
Comment (capetillo) [inline src/components/Project/ImageViewer.vue:51]: this as well and that's all :-)

**Review by mgdaily — APPROVED** [review]
Lookin' good to me! Just a couple of comments. Thanks for the TODOs - that helped a lot.

File: src/components/Project/LineChart.vue [hunk 3]
```diff
@@ -0,0 +1,113 @@
+<script setup>
```
[the remark this answers is not in the document]
Comment (capetillo) [reply src/components/Project/LineChart.vue:1]: yeah we can!

File: src/store/index.js [hunk 4]
```diff
@@ -70,6 +72,14 @@ export default createStore({
 
     setLargeImageCache(state, imageData) {
       state.largeImageCache = imageData
+    },
+
+    setRandomNumbers(state, numbers) {
```
[the remark this answers is not in the document]
Comment (capetillo) [reply src/store/index.js:77]: Yep, that and all of the store. I think that'll be one of next sprint's stories for me. Refactor from vuex store to pinia

**Pull request merged**